youtube_dl/extractor/pandoratv.py

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+from __future__ import unicode_literals
+
+import re
+import json
+
+from .common import InfoExtractor
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://(.*?\.)?channel.pandora.tv/channel/video.ptv\?.*?prgid=(?P<prgid>.*?)&.*'
```

Comment:
[Correct way](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/pluralsight.py#L95-L100) to implement order independent argument matching.